src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -708,5 +708,166 @@ public static nuint RotateRight(nuint value, int offset)
             return (nuint)RotateRight((uint)value, offset);
 #endif
         }
+
+        /// <summary>
+        /// Calculates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * uint32_t __crc32b (uint32_t a, uint8_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32(uint crc, byte data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // uint32_t __crc32b (uint32_t a, uint8_t b)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Runtime.Intrinsics.Arm.Crc32.IsSupported)
+            {
+                // uint32_t __crc32b (uint32_t a, uint8_t b)
+                return Runtime.Intrinsics.Arm.Crc32.ComputeCrc32(crc, data);
+            }
+
+
+            // Software fallback
```

Comment:
doesn't produce the same result as Sse42.Crc32 with `Crc32(1164749927, byte.MaxValue)`